File: leetcode/217-contains-duplicate.cpp
class Solution
{
 public:
  bool containsDuplicate(vector<int>& nums)
  {
    unordered_set<int> vec;
    for (int n : nums) {
      if (vec.find(n) != vec.end()) return true;
      vec.insert(n);
    }
    return false;
  }
};

File: leetcode/217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
  var dic = {};
  for (var i = 0; i < nums.length; ++i) {
    if (dic[nums[i]]) return true;
    dic[nums[i]] = true;
  }
  return false;
};
